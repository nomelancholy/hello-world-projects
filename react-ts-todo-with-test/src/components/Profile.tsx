import React, { useState } from "react";

interface Props {
  userName: string;
  name: string;
}

const Profile = ({ userName, name }: Props) => {

    const [inputText, setInputText] = useState('')

 const handleChange = (e:any) => {
     setInputText(e.target.value)
 }


  return (
    <div>
        <input type="text" onChange={handleChange} value={inputText} placeholder="anything"/>
      <span>
        <b>{userName}</b>
      </span>
      <span>{name}</span>
    </div>
  );
};

export default Profile;
