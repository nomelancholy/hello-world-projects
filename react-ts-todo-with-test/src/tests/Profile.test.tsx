import React from "react";
import { render, fireEvent } from "@testing-library/react";
import Profile from "../components/Profile";

describe("<Profile />", () => {
  it("matches snapshot", () => {
    const utils = render(<Profile userName="TakeKnowledge" name="jae-uk" />);
    console.log('utils', utils)
    expect(utils.container).toMatchSnapshot();
  });

  it("input text", () => {
    const utils = render(<Profile userName="TakeKnowledge" name="jae-uk" />);
    const input = utils.getByPlaceholderText("anything");
    fireEvent.change(input, { target: { value: "hello" } });
  });
});
