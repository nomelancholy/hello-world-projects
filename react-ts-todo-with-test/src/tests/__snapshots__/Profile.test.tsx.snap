// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Profile /> matches snapshot 1`] = `
<div>
  <div>
    <input
      placeholder="anything"
      type="text"
      value=""
    />
    <span>
      <b>
        TakeKnowledge
      </b>
    </span>
    <span>
      jae-uk
    </span>
  </div>
</div>
`;
